packages/smooth_app/lib/pages/prices/prices_proofs_page.dart

```diff
@@ -29,15 +31,94 @@ class PricesProofsPage extends StatefulWidget {
 
 class _PricesProofsPageState extends State<PricesProofsPage>
     with TraceableClientMixin {
-  late final Future<MaybeError<GetProofsResult>> _results = _download();
-
   static const int _columns = 3;
   static const int _rows = 5;
   static const int _pageSize = _columns * _rows;
 
+  late final InfiniteScrollProofManager _proofManager;
+  String? _bearerToken;
+  final ScrollController _gridScrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _proofManager = InfiniteScrollProofManager(
+      initialItems: const <Proof>[],
+      bearerTokenCallback: () => _bearerToken,
+      onAuthenticateNeeded: _authenticate,
```

Comment:
Not sure it's the page's job to deal with the token, but let's focus on the easier classes first.